src/marketplace/MakeATankSaleView.js

```diff
@@ -0,0 +1,122 @@
+//@flow strict
+import * as React from 'react';
+import OptionClass from '../armory/OptionClass.js';
+import {getAllUsersTanks} from '../globalComponents/tankAPIIntegration.js';
+import {getUser} from '../globalComponents/userAPIIntegration.js';
+import {makeASale} from './marketPlaceAPIConnections.js';
+
+type Props = {||}; 
+type State = {|
+	userId: string,
+	salePrice: number,
+	tankBeingSoldId: string,
+	itemAmount: number,
+	tanksToSell: Array<OptionClass>,
```

Comment:
Please make this an array of tanks, instead of an array the things that should be rendered. Like I mentioned on my previous PR review, your state should represent the data you are rendering, not the way you render it. I'm happy to talk with you about why this is better design if you would like.

Reply:
It will probably be helpful to talk to Adam since he wrote a bunch of code to do this.